botnim/sync.py

```diff
@@ -52,115 +52,106 @@ def openapi_to_tools(openapi_spec):
         if assistant.name == assistant_name:
             assistant_id = assistant.id
+            existing_assistant = assistant
+            logger.info(f"Found existing assistant: {assistant_id}")
             break
-    print(f'Assistant ID: {assistant_id}')
-    asst_params = dict(
-        name=assistant_name,
-        description=config['description'],
-        model='gpt-4o',
-        instructions=config['instructions'],
-        temperature=0.00001,
-    )
+
+    # Create new assistant
+    assistant_params = {
+        'name': assistant_name,
+        'description': config['description'],
+        'model': 'gpt-4o',
+        'instructions': config['instructions'],
+        'temperature': 0.00001,
+    }
+    
+    # Handle tools configuration
+    tools = []
     if config.get('tools'):
-        tools = tools or []
         for tool in config['tools']:
             if tool == 'code-interpreter':
-                tools.append(dict(type='code_interpreter'))
+                tools.append({"type": "code_interpreter"})
             else:
                 openapi_spec = (SPECS / 'openapi' / tool).with_suffix('.yaml').open()
                 openapi_spec = yaml.safe_load(openapi_spec)
                 openapi_tools = openapi_to_tools(openapi_spec)
-                # print(f'OpenAPI Tool: {tool}')
                 tools.extend(openapi_tools)
+    
     if tools:
-        asst_params['tools'] = tools
-    if tool_resources:
-        asst_params['tool_resources'] = tool_resources
-    import pprint
-    pprint.pprint(asst_params)
+        assistant_params['tools'] = tools
+    
     if assistant_id is None:
-        # Create a new assistant:
-        assistant = client.beta.assistants.create(**asst_params)
+        # Create new assistant
+        assistant = client.beta.assistants.create(**assistant_params)
         assistant_id = assistant.id
-        print(f'Assistant created: {assistant_id}')
-        # ...
+        logger.info(f'Created new assistant: {assistant_id}')
+        replace_context = True
+        logger.info('Forcing replace_context=True for new assistant')
     else:
-        # Update the existing assistant:
-        assistant = client.beta.assistants.update(assistant_id, **asst_params)
-        print(f'Assistant updated: {assistant_id}')
-        # ...
+        # Update existing assistant's configuration
+        assistant = client.beta.assistants.update(
+            assistant_id=assistant_id,
+            **assistant_params
+        )
+        logger.info(f'Updated assistant configuration: {assistant_id}')
+    
+    # Set up context if configured
+    vector_store_id = None
+    if config.get('context'):
+        logger.info(f"Context configuration found, replace_context={replace_context}")
+        if replace_context:
+            logger.info("Setting up new contexts...")
+            vector_store_id = context_manager.setup_contexts(config['context'])
+            
+            # Update assistant with new vector store
```

Comment:
Again, this entire bit of code makes much more sends to happen *before* we start working on the assistant. The only thing we need to update are the `tools` and `tool_resources` and let the assistant code will handle the create/update sequence.